### PR TITLE
[JENKINS-65657] Update xstream to 1.4.17

### PR DESCRIPTION
**Pull request by MRamonLeon**
See [JENKINS-65657](https://issues.jenkins-ci.org/browse/JENKINS-65657).
* https://x-stream.github.io/news.html#1.4.17
* https://github.com/x-stream/xstream/compare/XSTREAM_1_4_16...XSTREAM_1_4_17

### Proposed changelog entries

* Entry 1: Update from xstream 1.4.16 to 1.4.17.  See changelog at https://x-stream.github.io/changes.html#1.4.17

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by rsandell — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Review by car-roll — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by MarkEWaite** [comment]
I don't know how to assess the comment from the "Stream compatibility" section of the 1.4.17 changelog, but I assume it does not affect Jenkins.  It says:

> The following types are now blacklisted by default and the deserialization of XML containing one of the two types will fail. You will have to enable these types by explicit configuration, if you need them:
> * any type in the java.rmi.* and sun.rmi.* package hierarchies
> * the individual type com.sun.jndi.toolkit.dir.LazySearchEnumerationImpl

There are no `java.rmi` or `sun.rmi` references in Jenkins core or in the remoting library.

**Review by olamy — APPROVED** [review]

**Pull request merged**
